examples/ConsumptionSaving/example_ConsRiskyContribModel.py

```diff
@@ -0,0 +1,364 @@
+
+# Rebalancing
+plotSlices3D(
+    dfracFunc_Adj,
+    0,
+    mMax,
+    y_slices=n_slices,
+    y_name="n",
+    titles=["t = " + str(t) for t in periods],
+    ax_labs=["m", "d"],
+)
+# Share
+plotSlices3D(
+    ShareFuncSha,
+    0,
+    mMax,
+    y_slices=n_slices,
+    y_name="n",
+    titles=["t = " + str(t) for t in periods],
+    ax_labs=["m", "S"],
+)
+
+# Consumption
+shares = [0.0, 0.9]
+plotSlices4D(
+    cFuncFxd,
+    0,
+    mMax,
+    y_slices=n_slices,
+    w_slices=shares,
+    slice_names=["n_til", "s"],
+    titles=["t = " + str(t) for t in periods],
+    ax_labs=["m_til", "c"],
+)
+
+# %%
+# Solve a short, finite horizon version
+par_finite = init_risky_contrib.copy()
+
+# Four period model
+par_finite["PermGroFac"] = [2.0, 1.0, 0.1, 1.0]
+par_finite["PermShkStd"] = [0.1, 0.1, 0.0, 0.0]
+par_finite["TranShkStd"] = [0.2, 0.2, 0.0, 0.0]
+par_finite["AdjustPrb"] = [0.5, 0.5, 1.0, 1.0]
+par_finite["tau"] = [0.1, 0.1, 0.0, 0.0]
+par_finite["LivPrb"] = [1.0, 1.0, 1.0, 1.0]
+par_finite["T_cycle"] = 4
+par_finite["T_retire"] = 0
+par_finite["T_age"] = 4
+
+# Adjust discounting and returns distribution so that they make sense in a
+# 4-period model
+par_finite["DiscFac"] = 0.95 ** 15
+par_finite["Rfree"] = 1.03 ** 15
+par_finite["RiskyAvg"] = 1.08 ** 15  # Average return of the risky asset
+par_finite["RiskyStd"] = 0.20 * np.sqrt(15)  # Standard deviation of (log) risky returns
+
+
+# Create and solve
+ContribAgent = RiskyContribConsumerType(**par_finite)
+print("Now solving")
+t0 = time()
+ContribAgent.solve()
+t1 = time()
+print("Solving took " + str(t1 - t0) + " seconds.")
+
+# Plot Policy functions
+periods = [0, 2, 3]
+
+dfracFunc_Adj = [
+    ContribAgent.solution[t].stage_sols["Reb"].dfracFunc_Adj for t in periods
+]
+ShareFuncSha = [
+    ContribAgent.solution[t].stage_sols["Sha"].ShareFunc_Adj for t in periods
+]
+cFuncFxd = [ContribAgent.solution[t].stage_sols["Cns"].cFunc for t in periods]
+
+# Rebalancing
+plotSlices3D(
+    dfracFunc_Adj,
+    0,
+    mMax,
+    y_slices=n_slices,
+    y_name="n",
+    titles=["t = " + str(t) for t in periods],
+    ax_labs=["m", "d"],
+)
+# Share
+plotSlices3D(
+    ShareFuncSha,
+    0,
+    mMax,
+    y_slices=n_slices,
+    y_name="n",
+    titles=["t = " + str(t) for t in periods],
+    ax_labs=["m", "S"],
+)
+# Consumption
+plotSlices4D(
+    cFuncFxd,
+    0,
+    mMax,
+    y_slices=n_slices,
+    w_slices=shares,
+    slice_names=["n_til", "s"],
+    titles=["t = " + str(t) for t in periods],
+    ax_labs=["m_til", "c"],
+)
+
+# %%  Simulate the finite horizon consumer
+ContribAgent.track_vars = [
+    "pLvl",
+    "t_age",
+    "Adjust",
+    "mNrm",
+    "nNrm",
+    "mNrmTilde",
+    "nNrmTilde",
+    "aNrm",
+    "cNrm",
+    "Share",
+    "dfrac",
+]
+ContribAgent.T_sim = 4
+ContribAgent.AgentCount = 10
+ContribAgent.initialize_sim()
+ContribAgent.simulate()
+
+# %% Format simulation results
+
+import pandas as pd
+
+Data = ContribAgent.history
```

Comment:
more python idiomatic to keep these lower case

Reply:
i.e data not Data unless it's the name of a class

Reply:
Addressed in https://github.com/econ-ark/HARK/pull/832/commits/2d9ab76e9d0b299a87cc567c41ebece57fb8f8f4.

Thank you!